### PR TITLE
CORE-15756 Segregated and Granular Corda Workers

### PR DESCRIPTION
**Pull request by mbrkic-r3**
### Overview:
Following logistical changes were implemented (components and messaging transport used for their communication stay the same):
- CORE-16243: Created `PersistenceProcessor` that uses `LedgerPersistenceService` and `FlowPersistenceService` (they are not used anymore in DB worker).
- CORE-16243: Created `PersistenceWorker` that uses `PersistenceProcessor`.
- CORE-16228: Split out the `VerificationProcessor` from the `FlowWorker` into its own worker: a `VerificationWorker`
- CORE-16244: Created `FlowMapperProcessor` that uses `FlowMapperService` and `P2PFilterService` (they are no longer used in Flow Worker).
- CORE-16244: Created `FlowMapperWorker` that uses `FlowMapperProcessor`.

### Related PRs:
- https://github.com/corda/corda-api/pull/1227
- https://github.com/corda/corda-shared-build-pipeline-steps/pull/1108
- https://github.com/corda/corda-e2e-tests/pull/212
- https://github.com/corda/corda-non-functional-test/pull/227
- https://github.com/corda/corda5-enterprise/pull/85
- https://github.com/corda/corda-soak-test-deployment/compare/CORE-15756/seg-workers?expand=1 (Creating PR will disable Jenkins job https://ci02.dev.r3.com/job/Corda5/job/corda-resiliency-tests/job/CORE-15756%252Fseg-workers/)

### Additional Testing:
- ✅ Resiliency Tests: https://ci02.dev.r3.com/job/Corda5/job/corda-resiliency-tests/job/CORE-15756%252Fseg-workers/4/
- ✅ E2E Tests: https://ci02.dev.r3.com/job/Corda5/job/corda5-e2e-tests/job/PR-212/7/

### :exclamation: Merging strategy:
- Agreed with BLT to:

1. Merge this `corda-runtime-os` PR that uses feature branch of `corda-shared-build-pipeline-steps` (`CORE-15756/segregated-workers`)
2. Disable deletion of merged branches in `corda-shared-build-pipeline-steps`  and merge `corda-shared-build-pipeline-steps` feature branch (`CORE-15756/segregated-workers`)
3. Merge `corda-runtime-os` that uses 5.1 branch of `corda-shared-build-pipeline-steps` (that will be a new PR)
4. Enable deletion of merged branches in `corda-shared-build-pipeline-steps`  and delete `corda-shared-build-pipeline-steps` feature branch (`CORE-15756/segregated-workers`)

- It's hard to fix `corda5-enterprise` build (some dependencies are injected dynamically and in the end some of them use 5.1 branch), so related PR should be built (and merged) in the end, after everything else is merged.

### CODE REVIEW

**Review by ben-millar — DISMISSED** [review]

**Comment by mcgovc** [comment]
See comment - https://github.com/corda/corda-shared-build-pipeline-steps/pull/1108#issuecomment-1700897299

**Review by wzur-r3 — DISMISSED** [review]
LGTM

**Review by davidcurrie — COMMENTED** [review]
LGTM for @corda/cluster-management - will approve once TODOs are resolved

File: charts/corda/values.yaml [hunk 1]
```diff
@@ -980,6 +1116,86 @@ workers:
...
+      # -- run persistence worker with profiling enabled
+      enabled: false
+    # resource limits and requests configuration for the persistence worker containers.
+    resources:
+      # -- the CPU/memory resource requests for the persistence worker containers
+      requests: { }
+      # -- the CPU/memory resource limits for the persistence worker containers
+      limits: { }
+    # persistence worker JDBC connection pool configuration for cluster DB
+    clusterDbConnectionPool:
```
Comment (simon-johnson-r3) [inline charts/corda/values.yaml:1157]: We may find this triggers some database connection errors in pipelines. This is not necessarily the end of the world as our dev-prereqs still has the max allowed connections set quite low and we can increase it further. Just a heads up.

**Review by simon-johnson-r3 — DISMISSED** [review]

**Review by thiagoviana — DISMISSED** [review]
LGTM

**Review by davidcurrie — DISMISSED** [review]
LGTM

**Review by davidcurrie — APPROVED** [review]
Still LGTM!

**Review by thiagoviana — APPROVED** [review]

**Review by ben-millar — APPROVED** [review]

**Review by simon-johnson-r3 — APPROVED** [review]

**Review by mcgovc — APPROVED** [review]
LGTM

**Review by dickon — APPROVED** [review]
LGTM.

I think the justification in CORE-15756 is rather one sided, but if that's what we want this seems a nice piece of work to deliver it.

File: processors/flow-mapper-processor/src/main/java/net.corda.processors.flow.mapper/package-info.java
Comment (blsemo) [inline processors/flow-mapper-processor/src/main/java/net.corda.processors.flow.mapper/package-info.java:1]: The path on this one looks wrong! What happened when creating this package?

**Review by blsemo — DISMISSED** [review]
I am ok with the content of this change, however the path for the FlowMapperProcessor files look wrong - the packages should be represented as a directory hierarchy, not a directory with loads of dots in the name.

**Review by jujoramos — APPROVED** [review]
LGTM

**Review by josephzunigadaly — APPROVED** [review]
Rest owned files look ok

**Review by owenstanford — APPROVED** [review]
Looks ok to me

File: processors/flow-mapper-processor/src/main/java/net.corda.processors.flow.mapper/package-info.java
Comment (ben-millar) [reply processors/flow-mapper-processor/src/main/java/net.corda.processors.flow.mapper/package-info.java:1]: Fixed here: https://github.com/corda/corda-runtime-os/pull/4569

**Pull request merged**